redis/asyncio/connection.py

```diff
@@ -227,6 +227,10 @@ def __init__(
             _cache = None
         self.client_cache = client_cache if client_cache is not None else _cache
         if self.client_cache is not None:
+            if self.protocol not in [3, "3"]:
+                raise RedisError(
+                    "client caching is only supported with protocol version 3 or higher"
```

Comment:
@chayim As I remember we agreed to force RESP3 instead of raising an error? Error should be thrown after it